### PR TITLE
List support: `remove` method

### LINKED ISSUE #1693: remove: method removes the first occurrence of the element with the specified value.
### Feature Description
Add support for remove() method for lists in the Python translation.

### Test Code
This code should work when this issue is fixed:
```python

lis = [1, 2, 3]
lis.remove(2)
print(lis)
>> [1, 3]
```

### Proposed Solution
See description in https://github.com/pyccel/pyccel/issues/1529

### PR DESCRIPTION
**Pull request by Farouk-Echaref**
Add support for the `remove()` method of Python lists to the semantic stage and Python codegen. This fixes #1693.

**Commit Summary**

- Add a `ListRemove` subclass of `ListMethod`, representing the `remove()` method. The constructor checks for compatibility between the type of the `remove()` argument and the type of the list, and raises an error if they are not homogeneous.
- Add success and error tests for the `remove()` method involving many scenarios where the function could be used.

### CODE REVIEW

**Review by mustapha-belbiad — APPROVED** [review]
Looks good.

**Review by EmilyBourne — APPROVED** [review]
Looks good to me too

File: pyccel/ast/builtin_methods/list_methods.py [hunk 1]
```diff
@@ -229,3 +230,43 @@ class ListExtend(ListMethod):
 
     def __init__(self, list_obj, iterable) -> None:
         super().__init__(list_obj, iterable)
+
+#==============================================================================
+class ListRemove(ListMethod) :
+    """
+    Represents a call to the .remove() method.
+    
+    Represents a call to the .remove() method which removes the first occurence
```
Comment (yguclu) [inline pyccel/ast/builtin_methods/list_methods.py:239]: Typo: occurence -> occurrence

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good!

**Review by yguclu — APPROVED** [review]

**Pull request merged**